docs/api-reference/operator.istio.io.md

```diff
@@ -0,0 +1,3242 @@
+
+
+#### CNIGlobalConfig
+
+
+
+CNIGlobalConfig is a subset of the Global Configuration used in the Istio CNI chart.
+
+
+
+_Appears in:_
+- [CNIValues](#cnivalues)
+
+| Field | Description | Default | Validation |
+| --- | --- | --- | --- |
+| `defaultResources` _[ResourceRequirements](https://kubernetes.io/docs/reference/generated/kubernetes-api/v/#resourcerequirements-v1-core)_ | Default k8s resources settings for all Istio control plane components.<br /><br />See https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#resource-requests-and-limits-of-pod-and-container |  |  |
+| `hub` _string_ | Specifies the docker hub for Istio images. |  |  |
+| `imagePullPolicy` _[PullPolicy](https://kubernetes.io/docs/reference/generated/kubernetes-api/v/#pullpolicy-v1-core)_ | Specifies the image pull policy for the Istio images. one of Always, Never, IfNotPresent.<br />Defaults to Always if :latest tag is specified, or IfNotPresent otherwise. Cannot be updated.<br /><br />More info: https://kubernetes.io/docs/concepts/containers/images#updating-images |  |  |
+| `imagePullSecrets` _string array_ |  |  |  |
+| `logAsJson` _boolean_ |  |  |  |
+| `logging` _[GlobalLoggingConfig](#globalloggingconfig)_ | Specifies the global logging level settings for the Istio CNI component. |  |  |
+| `tag` _[IntOrString](https://kubernetes.io/docs/reference/generated/kubernetes-api/v/#intorstring-intstr-util)_ | Specifies the tag for the Istio CNI image. |  | XIntOrString: \{\} <br /> |
+| `variant` _string_ |  |  |  |
+
+
+#### CNIRepairConfig
+
+
+
+
+
+
+
+_Appears in:_
+- [CNIConfig](#cniconfig)
+
+| Field | Description | Default | Validation |
+| --- | --- | --- | --- |
+| `enabled` _boolean_ | Controls whether repair behavior is enabled. |  |  |
+| `hub` _string_ | Hub to pull the container image from. Image will be `Hub/Image:Tag-Variant`. |  |  |
+| `tag` _[IntOrString](https://kubernetes.io/docs/reference/generated/kubernetes-api/v/#intorstring-intstr-util)_ | The container image tag to pull. Image will be `Hub/Image:Tag-Variant`. |  | XIntOrString: \{\} <br /> |
+| `image` _string_ | Image name to pull from. Image will be `Hub/Image:Tag-Variant`.<br />If Image contains a "/", it will replace the entire `image` in the pod. |  |  |
+| `labelPods` _boolean_ | The Repair controller has 3 modes (labelPods, deletePods, and repairPods). Pick which one meets your use cases. Note only one may be used.<br />The mode defines the action the controller will take when a pod is detected as broken.<br />If labelPods is true, the controller will label all broken pods with <brokenPodLabelKey>=<brokenPodLabelValue>.<br />This is only capable of identifying broken pods; the user is responsible for fixing them (generally, by deleting them).<br />Note this gives the DaemonSet a relatively high privilege, as modifying pod metadata/status can have wider impacts. |  |  |
```

Comment:
The descriptions is not correct, is mentioning the description from reparPods

Reply:
@luksa

Reply:
This will be easy to fix, we just need to update the values in the go files referenced. But we will need to check every field description to check that matches the correct description

Reply:
@luksa we copy this from here https://github.com/istio/istio/blob/master/operator/pkg/apis/istio/v1alpha1/values_types.pb.go? Upstream has the same issue with the description of the fields. I'll submit a PR to fix those descriptions upstream

Reply:
I just saw that the description has a lot of references that are the same but at the end have the exact definition for the field

Reply:
There are three fields (labelPods, deletePods, and repairPods), and only one of them can be set at a time. That's why all three field descriptions start with "The Repair controller has 3 modes...". MAybe this description could be improved, but this doesn't have anything to do with this PR.